src/Compilers/CSharp/Test/Syntax/LexicalAndXml/PreprocessorTests.cs

```
@@ -2962,6 +2965,71 @@ private void CheckDiagnosticStringFileName(string compilationFileName, string li
             Assert.Equal(expectedErrorStringFileName, actualErrorStringFileName);
         }
 
+        [Fact]
+        public void TestErrorWithVersion()
+        {
+            var text = "#error version";
+            var node = Parse(text, SourceCodeKind.Regular);
+            TestRoundTripping(node, text, disallowErrors: false);
+            VerifyDirectivesSpecial(node, new DirectiveInfo
+            {
+                Kind = SyntaxKind.ErrorDirectiveTrivia,
+                Status = NodeStatus.IsActive,
+                Text = "version"
+            });
+
+            node.GetDiagnostics().Verify(
+                // (1,8): error CS1029: #error: 'version'
+                // #error version
+                Diagnostic(ErrorCode.ERR_ErrorDirective, "version").WithArguments("version").WithLocation(1, 8),
+                // (1,8): error CS8304: Compiler version: '42.42.42.42424 (<developer build>)'. Language version: 4.
+                // #error version
+                Diagnostic(ErrorCode.ERR_CompilerAndLanguageVersion, "version").WithArguments(GetExpectedVersion(), "4").WithLocation(1, 8)
+                );
+        }
+
+        [Fact]
+        public void TestErrorWithVersionNumber()
+        {
+            var text = "#error version:7.1";
+            var node = Parse(text, SourceCodeKind.Regular);
+            TestRoundTripping(node, text, disallowErrors: false);
+            VerifyDirectivesSpecial(node, new DirectiveInfo
+            {
+                Kind = SyntaxKind.ErrorDirectiveTrivia,
+                Status = NodeStatus.IsActive,
+                Text = "7.1"
```

Indeed, the verification method was silently dropping some expectations :-S